.github/workflows/firmware.yml

```diff
@@ -0,0 +1,17 @@
+name: Build Components
+
+on: [pull_request]
+
+jobs:
+  build_components:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v3.1.0
+      - name: Build TEL
+        uses: xanderhendriks/action-build-stm32cubeide@v13.0
+        with:
```

Comment:
question: will we be extending this functionality to all other boards in the future or just tel for now?

Reply:
Yes, the plan is that the new boards (DRD, STR, etc...) will be added to this workflow. Old boards (MCB, DID) will not be added.